cuda/CMakeLists.txt

```diff
@@ -128,6 +128,33 @@ target_sources(ginkgo_cuda
 foreach(source_file IN LISTS GKO_CUDA_COMMON_SOURCES)
     set_source_files_properties(${source_file} PROPERTIES LANGUAGE CUDA)
 endforeach(source_file)
+if(GINKGO_JACOBI_FULL_OPTIMIZATIONS)
+    # keep in sync with jacobi_common.hpp
+    set(GKO_CUDA_JACOBI_BLOCK_SIZES)
+    foreach(blocksize RANGE 1 32)
+        list(APPEND GKO_JACOBI_BLOCK_SIZES ${blocksize})
+    endforeach()
+else()
+    # keep in sync with jacobi_common.hpp
+    set(GKO_CUDA_JACOBI_BLOCK_SIZES 1 2 4 8 13 16 32)
```

Comment:
Maybe we should make a CMAKE define just for this so that these magical values are actually in sync? It can be common for HIP as well. You would need to add a definition with commas though

Reply:
Good idea, I generate the jacobi_common.hpp automatically now